Lib/glyphsLib/builder/custom_params.py

```diff
@@ -611,28 +619,58 @@ def to_ufo(self, builder, glyphs, ufo):
 register(GlyphOrderParamHandler())
 
 
-# See https://github.com/googlefonts/glyphsLib/issues/214
+# Handling of Glyph's (Pre)Filters is complicated. ufo2ft grew filter modules to mimic
+# some of Glyph's automatic features, but due to the impendance mismatch between the
+# flow of data in Glyphs and in UFOs plus Designspaces, they need to be handled in two
+# ways: once for filters that should be applied to masters and once for filters on
+# instances, which should be applied only to interpolated UFOs:
+#
+#   +------+
+#   |GSFont+-------------------+
+#   +----+-+                   |
+#        |                     |
+#      +-+-----------+       +-+----------+
+#      |GSFontMaster |       |GSIntance   |
+#      +-------------+       +------------+
+#       userData                    customParameters
+#         com...ufo2ft.filters        Filter & PreFilter
+#
+#            ^  |                      |  ^
+# roundtrips |  |                      |  |
+#            |  v                      |  |
+#        lib                           |  | roundtrips
+#          com...ufo2ft.filters        |  |
+#      +-----------+                   v  |
+#      |Master UFO |          lib
+#      +---+-------+            com.schriftgestaltung.customParameter...
+#          |
+#      +---+-----+        +----------+                    +-----------------+
+#      | Source  |        | Instance |    ------------>   |Interpolated UFO |
+#      +---+-----+        +-----+----+                    +-----------------+
+#          |                    |          goes 1 way        lib
+#  +-------+-----+              |     apply_instance_data()    com...ufo2ft.filters
+#  | Designspace +--------------+
+#  +-------------+
+#
+# The ufo2ft filters should roundtrip as-is between UFO source masters and GSFontMaster,
+# because that's how we use them in the UFO workflow with 1 master UFO = 1 final font
+# with filters applied.
+#
+# The Glyphs filters defined on GSInstance should keep doing what they were doing
+# already:
+#
+#  - first be copied as-is into the designspace instance's lib, which should roundtrip
+#    back to Glyphs
+#  - then be converted to ufo2ft equivalents and put in the final interpolated UFOs
+#    before they are compiled into final fonts. Those should not roundtrip because the
+#    interpolated UFO is discarded after compilation.
+#
```

Comment:
Maybe add that the param handler below handles only the "goes 1 way" case from the diagram